control-plane-operator/hostedclusterconfigoperator/controllers/resources/kas/admissionpolicies.go

```diff
@@ -0,0 +1,152 @@
+package kas
+
+import (
+	"context"
+	"fmt"
+
+	configv1 "github.com/openshift/api/config/v1"
+	operatorv1alpha1 "github.com/openshift/api/operator/v1alpha1"
+	hyperv1 "github.com/openshift/hypershift/api/hypershift/v1beta1"
+	"github.com/openshift/hypershift/control-plane-operator/hostedclusterconfigoperator/controllers/resources/manifests"
+	"github.com/openshift/hypershift/support/config"
+	"github.com/openshift/hypershift/support/upsert"
+	k8sadmissionv1beta1 "k8s.io/api/admissionregistration/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/ptr"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type AdmissionPolicy struct {
+	Name             string
+	MatchConstraints *k8sadmissionv1beta1.MatchResources
+	Validations      []k8sadmissionv1beta1.Validation
+}
+
+const (
+	AdmissionPolicyNameConfig = "config"
+	AdmissionPolicyNameICSP   = "icsp"
+)
+
+var (
+	defaultAdmissionPoliciesOperations = []k8sadmissionv1beta1.OperationType{"UPDATE", "DELETE"}
+	defaultMatchResourcesScope         = k8sadmissionv1beta1.ScopeType("*")
+	defaultMatchPolicyType             = k8sadmissionv1beta1.Equivalent
+	HCCOUserValidation                 = k8sadmissionv1beta1.Validation{
+		Expression: fmt.Sprintf("request.userInfo.username.contains('system:%s') || object.spec == oldObject.spec", config.HCCOUser),
```

Comment:
why do we do a `contains` check here instead of an equality check?
With this, a user with the username `system:hosted-cluster-config-operator-fake` for example, would be allowed to edit these resources, no?

Reply:
I would say yes, I can change it to be exactly that user.